### DISCUSSION TITLE
Missing id_shop when call hook actionUpdateQuantity in one case in multistore

### DISCUSSION THREAD

**Discussion by ComonSoft** [PrestaShop/PrestaShop#33155] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In stock manager StockAvailable class, when you are in multishop and call the method setQuantity(), the id_shop parameter is missing when calling hook actionUpdateQuantity.

### Expected behavior

The id_shop should be passed in parameters as is it in other calls in this class.

```
Hook::exec(
    'actionUpdateQuantity',
        [
            'id_product' => $id_product,
            'id_product_attribute' => $id_product_attribute,
            'quantity' => $quantity,
            'id_shop' => $id_shop,
        ]
    );

```

### Steps to reproduce

1. Create a module with a hook actionUpdateQuantity
2. Edit a product
3. Change stock value
4. Save
5. In hookActionUpdateQuantity in the module id_shop is not present in $parameters array

### PrestaShop version(s) where the bug happened

1.7.6.4 to 8.0.4

### PHP version(s) where the bug happened

PHP 7.4

### Your company or customer's name goes here (if applicable).

Com'onSoft

**Comment by AureRita** [comment]
Hi @ComonSoft 

Thank you for your report, I tried to reproduce your issue as you can see : 

[Untitled_ Jun 22, 2023 5_19 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/4c83349a-1f1b-4f0c-8cb3-32dc026f6e0d)


but currently i think we didn't trigger the hook, 
Did I miss something ? 

Waiting for your feedback

**Comment by kpodemski** [comment]
Hello @ComonSoft 

It's more like a feature request than a bug. I'll transfer this Issue to Discussions.

Maybe you'd like to contribute with a PR? If so, you can do that for 8.1.x branch 👍🏻

**Comment by Hlavtox** [comment]
@ComonSoft id_shop is passed to every call in 8.1 now. :-)
